### ISSUE TITLE
Elements with attached DOM events inside vg-if blocks stop firing when the said blocks are re-rendered.

### ISSUE THREAD

**Issue by dcs76** [vugu/vugu#99]
**Describe the bug**
When switching back and forth between 'views' selected by using vg-if the DOM events of re-visited views won't fire anymore.

**Software Versions**
Vugu version: v0.1.1-0.20191230174203-fe17e20eb6ce
Go version: 1.13.5
Browser and version: 79.0.3945.88 (Official Build) (64-bit) (Linux)

**To Reproduce**
The .vugu file below illustrates the issue:
```
<html>
    <head>
        <title>Test page</title>
    </head>
    <body>
        <div>
            <label vg-html='c.text'></label>
            <div vg-if='c.ActiveView == "view1"'>
                <input type="text" @keyup="c.Update(event)"/>
                <h3>This is view1</h3>
                <button @mouseup='c.SwitchView(event, "view2")'>Switch</button>
            </div>
            <div vg-if='c.ActiveView == "view2"'>
                <input type="text" @keyup="c.Update(event)"/>
                <h3>This is view2</h3>
                <button @mouseup='c.SwitchView(event, "view1")'>Switch</button>
            </div>
        </div>
        
    </body>
</html>

<script type="application/x-go">

type Root struct {
    ActiveView string `vugu:"data"`
    text string
}

var CurrentView = "view1"

func (c *Root) BeforeBuild() {
    c.ActiveView = CurrentView
}

func (c *Root) SwitchView(event *vugu.DOMEvent, viewname string) {
    fmt.Println("switching to "+viewname)
    go func() {
        ee := event.EventEnv()
        ee.Lock()
        CurrentView = viewname
        c.text = ""
        ee.UnlockRender()
    }()
}

func (c *Root) Update(event *vugu.DOMEvent) {
    fmt.Println("updating")
    c.text = event.PropString("target", "value")
}

</script>

```
**Expected behavior**
I expected the DOM events attached to buttons and input box to continue working after switching back to a previously rendered 'view'.

**Comment by bradleypeabody** [comment]
Thanks for reporting this.  I can reproduce it and will update when I have more info and/or a fix, hopefully soon.

**Comment by robertobatts** [comment]
I tried to debug it but I didn't come up with any solution. If it can be of any help, I initially thought taht for any reason the function was erased from VGNode.DOMEventHandlerSpecList, so I had a look into the structure of the nodes after the rendering but the function continues to be there.
I guess it has something to do with how the function are stored inside r.instructionList

**Comment by bradleypeabody** [comment]
Yeah I'm pretty sure this is related to the code that synchronizes the DOM and implements the instructionList on the JS side (see .../vugu/domrender/renderer-js-script.js).  I'm sure it's fixable and I hope to have a chance to look at this in more detail soon.

**Comment by dcs76** [comment]
Any news on this issue? Thanks.

**Comment by bradleypeabody** [comment]
@dcs76 Sorry for the delay in getting to this.  It is now fixed on the master branch.

The issue was that in cases where DOM elements were being deleted and then recreated the internal state of the renderer still thought the event listener was there.  I changed it so the listener is removed and re-added each render pass regardless and it's working as expected.  I also added a test case for this.  Should be all good now.
